ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncContext.java

```diff
@@ -125,31 +123,39 @@ public synchronized void updateHeads(final Hash head, final Hash finalizedBlockH
 
   public synchronized CompletableFuture<Void> syncBackwardsUntil(final Hash newBlockHash) {
     Optional<CompletableFuture<Void>> maybeFuture =
-        Optional.ofNullable(this.currentBackwardSyncFuture.get());
+        Optional.ofNullable(this.currentBackwardSyncStatus.get())
+            .map(status -> status.currentFuture);
     if (isTrusted(newBlockHash)) {
       return maybeFuture.orElseGet(() -> CompletableFuture.completedFuture(null));
     }
     backwardChain.addNewHash(newBlockHash);
     return maybeFuture.orElseGet(
         () -> {
-          CompletableFuture<Void> future = prepareBackwardSyncFutureWithRetry();
-          this.currentBackwardSyncFuture.set(future);
-          return future;
+          LOG.info("Starting a new backward sync session");
+          Status status = new Status(prepareBackwardSyncFutureWithRetry());
+          this.currentBackwardSyncStatus.set(status);
+          return status.currentFuture;
         });
   }
 
   public synchronized CompletableFuture<Void> syncBackwardsUntil(final Block newPivot) {
     Optional<CompletableFuture<Void>> maybeFuture =
-        Optional.ofNullable(this.currentBackwardSyncFuture.get());
+        Optional.ofNullable(this.currentBackwardSyncStatus.get())
+            .map(status -> status.currentFuture);
     if (isTrusted(newPivot.getHash())) {
       return maybeFuture.orElseGet(() -> CompletableFuture.completedFuture(null));
     }
     backwardChain.appendTrustedBlock(newPivot);
     return maybeFuture.orElseGet(
         () -> {
-          CompletableFuture<Void> future = prepareBackwardSyncFutureWithRetry();
-          this.currentBackwardSyncFuture.set(future);
-          return future;
+          LOG.info("Starting a new backward sync session");
+          LOG.info("Backward sync target block is {}", newPivot.toLogString());
```

Comment:
should these two log lines be combined?

Reply:
Not joining since a new session starts also in the previous method, where we still do not have the target block, so for keeping the log uniform I think it is better to keep the 2 lines separated